src/phoebus_utils/programming_utils.hpp

```diff
@@ -0,0 +1,49 @@
+// © 2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+
+#ifndef PHOEBUS_UTILS_PROGRAMMING_UTILS_HPP_
+#define PHOEBUS_UTILS_PROGRAMMING_UTILS_HPP_
+
+#include <algorithm>
+#include <vector>
+
+namespace programming {
+
+template <class SPECIAL, template <class...> class TEMPL>
+struct is_specialization_of : public std::false_type {};
+
+template <template <class...> class TEMPL, class... TPARAMS>
+struct is_specialization_of<TEMPL<TPARAMS...>, TEMPL> : public std::true_type {};
+
+template <typename T>
+bool soft_equiv(const T x, const T y, const T tolerance = static_cast<T>(1.e-10f)) {
+  bool equivalent = false;
+  if (2. * fabs(y - x) < tolerance * (std::fabs(x) + std::fabs(y))) {
+    equivalent = true;
+  }
+
+  return equivalent;
+}
+
+template <typename T>
+bool ContainedInVector(T &item, std::vector<T> &container) {
+  if (std::find(container.begin(), container.end(), item) == container.end()) {
+    std::stringstream msg;
+    return false;
+  }
+  return true;
+}
```

Comment:
Others may have other generally useful little gizmos worth putting in here? One nice thing might be a (host-side) class that maps between strings and enums, for e.g. safely converting from string inputs to internal enums.

Reply:
:+1: for programming gizmos. Wouldn't a class that did that have to be enum specific? Maybe not if we played evil macro games...

Reply:
I was thinking some kind of class that held like an `std:map<MyEnum, std::string>` internally or something. Maybe that doesn't work, I didn't work it all the way out.